firedrake/mesh.py

```diff
@@ -224,6 +224,36 @@ def facet_cell_map(self):
                        "facet_to_cell_map")
 
 
+@PETSc.Log.EventDecorator()
+def _from_netgen(ngmesh,comm=None):
+    """
+    Create a DMPlex from an Netgen mesh
+
+    :arg ngmesh: NetGen Mesh
+    """
+    if len(ngmesh.Elements3D()) != 0:
+        raise RuntimeError("Only 2D meshes are implemented at the moment.")
+    else:
+        V = [];
+        T = [];
+        for v in list(ngmesh.Points()):
+            V = V + [[float(v[0].__repr__()),float(v[1].__repr__())]]
+        for t in ngmesh.Elements2D():
+            T = T + [[int(v.__repr__())-1 for v in t.vertices]]
+    plex = _from_cell_list(2, T, V, comm, name=None)
+    for e in ngmesh.Elements1D():
+        vStart, vEnd = plex.getDepthStratum(0)   # vertices
+        join = plex.getJoin([vStart+int(v.__repr__())-1 for v in e.vertices])
+        plex.setLabelValue(dmcommon.FACE_SETS_LABEL, join[0], int(e.index))
+    return plex
+@PETSc.Log.EventDecorator()
+def ngLabels(ngmesh,labels):
```

Comment:
So sorry, but I don't understand what you mean by a Netgen functionality.
Netgen allows you to use string labels at the boundary and this little function allows you to specify the same labels used when defining the geometry via Netgen to impose boundary conditions.

Reply:
So, as far as I understand, `ngmesh.GetBCName(ID)` maps the boundary ID (`ID`) to the boundary name; it would be a lot nicer if we had the inverse of this map (say `ngmesh.GetBCIDs("boundary_name")`-> list of boundary IDs) available in Netgen. (As for current code, we should `unique(BID)`.

Reply:
Ah, now I get what you mean. I'm sorry I haven't yet had time to have a look at the Netgen source code. I'll try to add this feature next week.

Reply:
Added to netgen GetBCIDs, on my [fork](https://github.com/UZerbinati/netgen). I'll wait to add all the features needed for the firedrake interface and then open a pull request.